cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -52,6 +52,9 @@ type TestServerInstanceOptions struct {
 
 	// Enable cert-auth for the kube-apiserver
 	EnableCertAuth bool
+
+	// ModifyServerRunOptions allows for providing a custom function to modify ServerRunOption just before running the server
+	ModifyServerRunOptions func(opts *options.ServerRunOptions)
```

Comment:
I don't think we want this. The whole testing apiserver is meant to be driven through flags. There should not be a generic tweak func.

Reply:
I am going to remove this, as you mentioned we want to drive apiserver through flags.

Reply:
Please review new changes in this commit https://github.com/kubernetes/kubernetes/pull/93021/commits/d30d3903741bf0832d0bf6ca10def65f15b5dcf9

I have pre-generated the certs and removed the ModifyServerRunOptions